File: ts-koa/src/echo_service.ts
import * as service from './spec/echo_service'
import * as models from './spec/models'

export let echoService = (): service.EchoService => {
    let echoBodyString = async (params: service.EchoBodyStringParams): Promise<string> => {
        return params.body
    }

    let echoBody = async (params: service.EchoBodyParams): Promise<models.Message> => {
        return {int_field: params.body.int_field, string_field: params.body.string_field}
    }

    let echoQuery = async (params: service.EchoQueryParams): Promise<models.Message> => {
        return {int_field: params.int_query, string_field: params.string_query}
    }

    let echoHeader = async (params: service.EchoHeaderParams): Promise<models.Message> => {
        return {int_field: params['int-header'], string_field: params['string-header']}
    }
    
    let echoUrlParams = async (params: service.EchoUrlParamsParams): Promise<models.Message> => {
        return {int_field: params.int_url, string_field: params.string_url}
    }     

    let sameOperationName = async (): Promise<service.SameOperationNameResponse> => {
        return {status: 'ok'}
    }
    
    return {echoBodyString, echoBody, echoQuery, echoHeader, echoUrlParams, sameOperationName}
}

File: java-micronaut-moshi/src/main/java/test_service/services/echo/EchoService.java
package test_service.services.echo;

import java.math.BigDecimal;
import java.time.*;
import java.util.*;
import java.io.*;
import test_service.models.*;

public interface EchoService {
	String echoBodyString(String body);
	Message echoBodyModel(Message body);
	List<String> echoBodyArray(List<String> body);
	Map<String, String> echoBodyMap(Map<String, String> body);
	Parameters echoQuery(int intQuery, long longQuery, float floatQuery, double doubleQuery, BigDecimal decimalQuery, boolean boolQuery, String stringQuery, String stringOptQuery, String stringDefaultedQuery, List<String> stringArrayQuery, UUID uuidQuery, LocalDate dateQuery, List<LocalDate> dateArrayQuery, LocalDateTime datetimeQuery, Choice enumQuery);
	Parameters echoHeader(int intHeader, long longHeader, float floatHeader, double doubleHeader, BigDecimal decimalHeader, boolean boolHeader, String stringHeader, String stringOptHeader, String stringDefaultedHeader, List<String> stringArrayHeader, UUID uuidHeader, LocalDate dateHeader, List<LocalDate> dateArrayHeader, LocalDateTime datetimeHeader, Choice enumHeader);
	UrlParameters echoUrlParams(int intUrl, long longUrl, float floatUrl, double doubleUrl, BigDecimal decimalUrl, boolean boolUrl, String stringUrl, UUID uuidUrl, LocalDate dateUrl, LocalDateTime datetimeUrl, Choice enumUrl);
	EchoEverythingResponse echoEverything(Message body, float floatQuery, boolean boolQuery, UUID uuidHeader, LocalDateTime datetimeHeader, LocalDate dateUrl, BigDecimal decimalUrl);
	SameOperationNameResponse sameOperationName();
}
